src/python/pants/backend/terraform/dependency_inference.py

```diff
@@ -143,8 +147,44 @@ async def infer_terraform_module_dependencies(
     return InferredDependencies(terraform_module_addresses)
 
 
+@dataclass(frozen=True)
+class GetTerraformDependenciesRequest:
+    source_files: SourceFiles
+    directories: Tuple[str, ...]
+
+
+@dataclass(frozen=True)
+class TerraformDependencies:
+    fetched_deps: Tuple[Tuple[str, Digest], ...]
```

Comment:
This is `(dir, Digest of .terraform and .terraform.lock.hcl for that dir)` ? Worth documenting, as that is not at all intuitive.

Reply:
Also, are these files in fact relocatable/cacheable? Are they remote-cacheable?

Reply:
they're definitely cacheable. My concern is on the grain of the cache and the cache key. Currently I think they'd all be considered 1 big digest, and the cache key would be all TF files requested. Both can be made smaller. I talk a bit more about it in the comment at the head of the MR. We can extract the list of required providers and individually download them (we can then either have terraform install from these local sources or emplace them ourselves). Terraform also supports lockfiles, which we could in some way leverage.
I'm pretty sure they could be remote-cached as well.
The combined size can be quite large, ours are around 200M (I think the big cloud Providers are most of it)